src/GraphQL/TypeApi.hs

```diff
@@ -7,6 +7,7 @@
 {-# LANGUAGE TypeApplications #-}
 {-# LANGUAGE TypeInType #-}
 {-# LANGUAGE MultiParamTypeClasses #-}
+{-# LANGUAGE UndecidableInstances #-} -- for TypeError
```

Comment:
Without this we can't have nice TypeError messages. The worst case is that the type checker loops and dies when running out of stack. Compiling this code successfully ist the same as a unit test for this case. On top of that it's a closed type family so no one else can ever do anything to hit a type checker loop.